app/src/lib/databases/pull-request-database.ts

```diff
@@ -16,13 +17,14 @@ export interface IPullRequestRef {
   readonly sha: string
 }
 
-export interface IPullRequest {
+export interface IBasePullRequestRef extends IPullRequestRef {
   /**
-   * The database ID. This will be undefined if the pull request hasn't been
-   * inserted into the DB.
+   * The database ID of the GitHub repository in which this ref lives.
    */
-  readonly id?: number
+  readonly repoId: number
```

Comment:
I was initially hesitant to touch this area as the previous code seemed to have been designed with at least a partway intention to allow storing pull requests where the head repository had since been deleted. As such I was concerned that there might be users out there with databases containing null head refs. Then I realized that since we'll be recreating the table anyway this might be a good opportunity to make this change.

I've now made it mandatory to have both a head and a base branch for a repository to be stored and presented in Desktop. If we decide to support this fully in the future it's better to start on a clean slate anyway rather than build upon the shaky foundations we had before.